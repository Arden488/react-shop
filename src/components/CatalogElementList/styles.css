.product-list {
  display: grid;
  grid-template-columns: repeat(3, minmax(240px, 1fr));
  grid-gap: 10px;
}