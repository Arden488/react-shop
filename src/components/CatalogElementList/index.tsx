import * as React from 'react';

import CatalogElementListItem from '../CatalogElementListItem';

import './styles.css';

interface ICatalogElementListProps {
  items: number[]
}

class CatalogElementList extends React.Component<ICatalogElementListProps> {
  public render() {
    const items = this.props.items.map(item => this.renderListItem(item));

    return (
      <div className="product-list">
        {items}
      </div>
    );
  }
  
  private renderListItem(item: number) {
    return <CatalogElementListItem code="1" key={item} />;
  }
}

export default CatalogElementList;
