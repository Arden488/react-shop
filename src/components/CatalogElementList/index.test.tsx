import { shallow } from 'enzyme';
import * as React from 'react';

import CatalogElementList from './index';

let component: React.ReactNode = null;

beforeEach(() => {
  const items = [{
    article: 762015,
    id: 1,
    image: "iphone8_red.webp",
    price: "39990",
    title: "Apple iPhone 8 32 ГБ (PRODUCT)RED",
    type: "iphone7",
  }, {
    article: 762016,
    id: 2,
    image: "iphone8_red.webp",
    price: "49990",
    title: "Apple iPhone 8 64 ГБ (PRODUCT)RED",
    type: "iphone8",
  }, {
    article: 762017,
    id: 3,
    image: "iphone8_red.webp",
    price: "59990",
    title: "Apple iPhone 8 256 ГБ (PRODUCT)RED",
    type: "iphone7",
  }]
  component = shallow(<CatalogElementList items={items} />);
});

it('renders correctly', () => {
  expect(component).toMatchSnapshot();
});
