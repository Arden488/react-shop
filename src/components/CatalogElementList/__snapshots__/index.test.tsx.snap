// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="product-list"
>
  <CatalogElementListItem
    code="1"
    key="1"
  />
  <CatalogElementListItem
    code="1"
    key="2"
  />
  <CatalogElementListItem
    code="1"
    key="3"
  />
  <CatalogElementListItem
    code="1"
    key="4"
  />
  <CatalogElementListItem
    code="1"
    key="5"
  />
</div>
`;
